Linux - Workouts 1 - 3 Improvements (#2096)
enkidevs/curriculum: linux/user-and-file-management/linux-intro/hello-linux.md

---
author: kapnobatai136

aspects:
  - introduction

type: normal

category: must-know

tags:
  - introduction
  - linux

links:
  - '[Install Ubuntu](https://ubuntu.com/tutorials/tutorial-install-ubuntu-desktop#1-overview){website}'
  - '[Try Ubuntu](https://ubuntu.com/tutorials/try-ubuntu-before-you-install#1-getting-started){website}'

---

# Hello Linux

---
## Content

All this talk, and we still haven't shown you how to use Linux 😅.

> > Note: we're going to assume you have an installed Linux distribution. If you don't, check out the *Learn More* section for Ubuntu installation guides.

Reviewer: Sometimes you use Note and sometimes **Note** in these callouts.

We should be consistent with the font-weight but why not use an emoji instead?

One idea:

```suggestion
> 💡We're going to assume you have an installed Linux distribution. If you don't, check out the *Learn More* section for Ubuntu installation guides.
```

![printing-hello-linux-in-shell](https://img.enkipro.com/b293f5a296e067d80f9dc7e77c7072d3.gif)

This is the Linux terminal[1]! It's where you'll be typing most of the commands.

> In computer science, what you see in the example is called *printing*, a fancy word for displaying text. 

Reviewer: Maybe more precise to talk about the command? Especially since we mention it right before? 

Also, the idea of commands is much more important here than the detail that displaying text in a terminal is called "printing":

```suggestion
What you see in the example is how to use the `echo` command to display some text. 
```

> > **Note:** We'll explain how printing works in the next workout.

Reviewer: Why don't we explain the "shell"-"terminal" relationship part here? The next insight feels quite crowded without any other visual element besides text?

Reviewer: We should only do this if it's easy to keep the same quality of reading flow, but no need to push it.

Author: move the comparison here. the flow is better this way

Reviewer: Cool 👍 

Remember to mention that a shell isn't exactly the same as the terminal and that it's a program running in the terminal.

---
## Practice

In Linux, you type commands in the ???.

* terminal
* text editor app
* search bar

---
## Footnotes

[1:Terminal]

> This is what facilitates the communication between you and the computer. The term originates from the first computers, which had a physical terminal that was used to communicate. 

Reviewer: I find the word "enables" a bit more casual than "facilitates" 🤷‍♂️

Reviewer: What do you think of this?

```suggestion
Similarly to how you can use a mouse to tell the computer to open a file, you can also type in a command in the **terminal**. 

From opening files to increasing the computer volume, there's a terminal command for anything the computer can do.

The term "terminal" originates from the first computers, which had a physical terminal that was used to communicate. 
```
